doc/examples/type_hints.rst

```diff
@@ -97,28 +97,89 @@ You can use :py:class:`~typing.TypedDict` (Python 3.8+) when using a well-define
 These methods automatically add an "_id" field.
 
 .. doctest::
+  :pyversion: >= 3.8
 
   >>> from typing import TypedDict
   >>> from pymongo import MongoClient
   >>> from pymongo.collection import Collection
+  >>> from bson import ObjectId
   >>> class Movie(TypedDict):
   ...       name: str
   ...       year: int
   ...
   >>> client: MongoClient = MongoClient()
   >>> collection: Collection[Movie] = client.test.test
+  >>> collection.drop()
   >>> inserted = collection.insert_one(Movie(name="Jurassic Park", year=1993))
-  >>> result = collection.find_one({"name": "Jurassic Park"})
+  >>> result = collection.find_one({})
   >>> assert result is not None
   >>> assert result["year"] == 1993
   >>> # This will not be type checked, despite being present, because it is added by PyMongo.
   >>> assert type(result["_id"]) == ObjectId
```

Comment:
The phrases "will not be type checked" and "cannot be type checked" appear throughout this PR but it's not that these lines won't be typed checked, they will fail type checking. It might be clearer to express this by adding a type ignore comment:
```python
assert result["_id"]  # type:ignore[typeddict-item]
``` 

Separately, I think we can remove the ObjectId in this example, just accessing result["_id"] with the type:ignore seems like enough.